Content.Server/Implants/SubdermalImplantSystem.cs

```diff
@@ -107,42 +111,84 @@ private void OnScramImplant(EntityUid uid, SubdermalImplantComponent component,
             _pullingSystem.TryStopPull(ent, pull);
 
         var xform = Transform(ent);
-        var entityCoords = xform.Coordinates.ToMap(EntityManager, _xform);
+        var targetCoords = SelectRandomTileInRange(xform, implant.TeleportRadius);
 
-        // try to find a valid position to teleport to, teleport to whatever works if we can't
-        var targetCoords = new MapCoordinates();
-        for (var i = 0; i < implant.TeleportAttempts; i++)
+        if (targetCoords != null)
         {
-            var distance = implant.TeleportRadius * MathF.Sqrt(_random.NextFloat()); // to get an uniform distribution
-            targetCoords = entityCoords.Offset(_random.NextAngle().ToVec() * distance);
+            _xform.SetWorldPosition(ent, targetCoords.Value.Position);
+            _xform.AttachToGridOrMap(ent, xform);
```

Comment:
You already have the EntityCoordinates (you know the target grid and position) from SelectRandomTileInRange so just use that instead of sending out 2 moveevents + doing the matrix transform again (1 of which is redundant).